pkg/service/providerindex/providerindex.go

```diff
@@ -87,18 +87,22 @@ func (pi *ProviderIndexService) getProviderResults(ctx context.Context, mh mh.Mu
 	s.AddEvent("searching in cache")
 	res, err := pi.providerStore.Members(ctx, mh)
 	if err == nil {
-		s.AddEvent("cache hit")
-		return res, nil
-	}
-	if !errors.Is(err, types.ErrKeyNotFound) {
-		telemetry.Error(s, err, "fetching from cache")
-		return nil, err
-	}
+		res, _ = filterCodecs(res, targetClaims)
```

Comment:
if we need to filter results it may make sense to include the claim type in the key, store each result under its own key and request directly the keys we are interested in.